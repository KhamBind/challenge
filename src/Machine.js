import React from 'react'
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'
import * as machineAction from './store/machine'
import Health from './Health'

class Machine extends React.Component {
    state = {
        name: ''
    }
    handleChange = (e) => {
        this.setState({
            name: e.target.value
        })
    }
    handleClick() {
        this.props.actions.putMachine(this.props.machine.id, this.state.name)
        this.props.history.push('/machines');

    }
    componentDidMount() {
        let machineid = this.props.match.params.machineid;
        if (machineid) this.props.actions.getMachine(machineid);
    }
    render() {
        const {machine} = this.props;
        if (!machine) return(<div></div>);
        return(
            <div className="detail">
                <div className="left-side">
                    <h2>{machine.name}</h2>
                    <h3>Update Device</h3>
                    <h4>Name:</h4>
                    <input
                        placeholder={machine.name}
                        value={this.state.name}
                        onChange={this.handleChange}
                    />
                    <button onClick={() => this.handleClick()}>SUBMIT</button>
                </div>
                <div className="right-side">
                    <div className="center">
                        <h1> {machine.health} </h1>
                        <Health value={machine.health} />
                    </div>
                    <h3>Stats</h3>
                    <h4>IP Adress: {machine.ip_address}</h4>
                </div>
            </div>
        )    
    }
}

const mapStateToProps = (state) => ({machine: state.machine.machine});
const mapDispatchToProps = (dispatch) => ({
    actions: bindActionCreators({...machineAction}, dispatch)
})
export default connect(mapStateToProps, mapDispatchToProps)(Machine);

