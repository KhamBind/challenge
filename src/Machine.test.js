import React from 'react'
import { Provider } from 'react-redux'
import renderer from 'react-test-renderer'
import configureStore from 'redux-mock-store'
import Machine from './Machine'

const mockStore = configureStore([]);
describe('Machine React-Redux Component', () => {
    let store;
    let component;

    beforeEach(() => {
        store = mockStore({
            machine: {
            machine: {
                name: "Machine1",
                health: 50,
                ip_address: "127.0.0.1"
            },
        },
        });
        store.dispatch = jest.fn();
        component = renderer.create(
            <Provider store={store}>
                <Machine match={{params: {machineid: "123456789"}}} />
            </Provider>
        );
    });
    it('should render with given state from Redux store', () => {
        expect(component.toJSON()).toMatchSnapshot();
    });
});