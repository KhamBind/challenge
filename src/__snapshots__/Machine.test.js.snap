// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Machine React-Redux Component should render with given state from Redux store 1`] = `
<div
  className="detail"
>
  <div
    className="left-side"
  >
    <h2>
      Machine1
    </h2>
    <h3>
      Update Device
    </h3>
    <h4>
      Name:
    </h4>
    <input
      onChange={[Function]}
      placeholder="Machine1"
      value=""
    />
    <button
      onClick={[Function]}
    >
      SUBMIT
    </button>
  </div>
  <div
    className="right-side"
  >
    <div
      className="center"
    >
      <h1>
         
        50
         
      </h1>
      <div
        className="progress-bar"
      >
        <div
          className="progress-inbar"
          style={
            Object {
              "backgroundColor": "#5cb85c",
              "width": "50%",
            }
          }
        />
      </div>
    </div>
    <h3>
      Stats
    </h3>
    <h4>
      IP Adress: 
      127.0.0.1
    </h4>
  </div>
</div>
`;
